set.mm

```diff
@@ -776576,6 +776859,676 @@ separated by (open) neighborhoods.  See ~ sepnsepo for the equivalence
+                               ( x .<_ y <-> ( x H y ) =/= (/) ) ) $.
+    ${
+      catprslem.x $e |- ( ph -> X e. B ) $.
+      catprslem.y $e |- ( ph -> Y e. B ) $.
+      $( Lemma for ~ catprs .  (Contributed by Zhi Wang, 18-Sep-2024.) $)
+      catprslem $p  |- ( ph -> ( X .<_ Y <-> ( X H Y ) =/= (/) ) ) $=
+        ( vz vw cv wbr co c0 wne wb wral breq1 oveq1 neeq1d bibi12d breq2 oveq2
+        weq cbvral2vw sylib wcel wi wceq wa breq12 oveq12 rspc2gv syl2anc mpd )
+        ALNZMNZFOZUSUTEPZQRZSZMDTLDTZGHFOZGHEPZQRZSZABNZCNZFOZVJVKEPZQRZSZCDTBD
+        TVEIVOVDUSVKFOZUSVKEPZQRZSBCLMDDBLUGZVLVPVNVRVJUSVKFUAVSVMVQQVJUSVKEUBU
+        CUDCMUGZVPVAVRVCVKUTUSFUEVTVQVBQVKUTUSEUFUCUDUHUIAGDUJHDUJVEVIUKJKVDVIL
+        MGHDDUSGULUTHULUMZVAVFVCVHUSGUTHFUNWAVBVGQUSGUTHEUOUCUDUPUQUR $.
+    $}
+
+    catprs.b $e |- ( ph -> B = ( Base ` C ) ) $.
+    catprs.h $e |- ( ph -> H = ( Hom ` C ) ) $.
+    ${
+      catprs.c $e |- ( ph -> C e. Cat ) $.
+      $( A preorder can be extracted from a category.  The category might not
+         be thin.  The hypothesis catprs.1 can be obtained from ~ ralrimivva .
+         See ~ catprsc for a construction.  The conclusion can be converted to
+         the quantified form with ~ ralrimivvva to apply ~ isprs .  See
+         ~ catprs2 .  (Contributed by Zhi Wang, 18-Sep-2024.) $)
+      catprs $p |- ( ( ph /\ ( X e. B /\ Y e. B /\ Z e. B ) ) -> ( X .<_ X
+                  /\ ( ( X .<_ Y /\ Y .<_ Z ) -> X .<_ Z ) ) ) $=
+        ( wcel wa wbr co c0 wne w3a wi ccid cfv chom eqid adantr simpr1 eleqtrd
+        cbs ccat wceq catidcl oveqd eleqtrrd ne0d cv wb wral catprslem ad2antrr
+        mpbird eleq2d pm5.32i cco simplr1 simplr2 simplr3 simpr2 biimpa adantrr
+        3anbi123d eqnetrrd sylanbr simpr3 adantrl catcone0 sylanb eqnetrd jca
+        ex ) AHDOZIDOZJDOZUAZPZHHGQZHIGQZIJGQZPZHJGQZUBWFWGHHFRZSTWFWLHEUCUDZUD
+        ZWFWNHHEUEUDZRWLWFEUJUDZEWMWOHWPUFZWOUFZWMUFAEUKOZWENUGWFHDWPAWBWCWDUHZ
+        ADWPULWELUGUIUMWFFWOHHAFWOULZWEMUGUNUOUPWFBCDFGHHABUQZCUQZGQXBXCFRSTURC
+        DUSBDUSWEKUGZWTWTUTVBWFWJWKWFWJPZWKHJFRZSTZXEXFHJWORZSXEFWOHJAXAWEWJMVA
+        ZUNWFAHWPOZIWPOZJWPOZUAZPZWJXHSTAWEXMAWBXJWCXKWDXLADWPHLVCADWPILVCADWPJ
+        LVCVLVDZXNWJPWPEEVEUDZWOHIJWQWRXPUFAWSXMWJNVAXJXKXLAWJVFXJXKXLAWJVGXJXK
+        XLAWJVHXNWFWJHIWORZSTXOXEHIFRZXQSXEFWOHIXIUNWFWHXRSTZWIWFWHXSWFBCDFGHIX
+        DWTAWBWCWDVIZUTVJVKVMVNXNWFWJIJWORZSTXOXEIJFRZYASXEFWOIJXIUNWFWIYBSTZWH
+        WFWIYCWFBCDFGIJXDXTAWBWCWDVOZUTVJVPVMVNVQVRVSWFWKXGURWJWFBCDFGHJXDWTYDU
+        TUGVBWAVT $.
+
+      ${
+        $d B u v $.  $d C u v w $.  $d ph u v w $.
+        catprs2.l $e |- ( ph -> .<_ = ( le ` C ) ) $.
+        $( A category equipped with the induced preorder is a preordered set,
+           or a proset.  See ~ catprs for a more primitive version.  See
+           ~ prsthinc for constructing a thin category from a proset.
+           (Contributed by Zhi Wang, 18-Sep-2024.) $)
+        catprs2 $p |- ( ph -> C e. Proset ) $=
```

Comment:
I think this is the main theorem, more than catprs. The informal definition of the induced preorder that I wrote below could be written here too/instead.

Reply:
I don't like this being the main theorem because the construction is arbitrary. One can alternatively construct it as
``.<_ = { <. x , y >. | ( x H y ) =/= (/) }``, which looks nicer to me, to be honest, but loses the property of ``dom .<_ = B``. A consequence is that you cannot have ``(.<_ \ _I ) Po B``. It is still an element of ``( le " Proset )`` though, I believe.

By the way, where is the informal definition you wrote?

Reply:
Done in latest commits